File: 1293/1293.cs
public class Solution {
    public int ShortestPathInAGridWithObstacl(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(ShortestPathInAGridWithObstacl(root.left), ShortestPathInAGridWithObstacl(root.right));
    }
}

File: 1293/1293.go
func shortestPathInAGridWithObstacl(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := shortestPathInAGridWithObstacl(root.Left)
	r := shortestPathInAGridWithObstacl(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}